src/components/inputs/z-input/index.stories.mdx

```diff
@@ -1,89 +1,320 @@
+      html`
+        <z-input
+          type=${args.typeText}
+          label=${args.label}
+          aria-label=${args.ariaLabel}
+          placeholder=${args.placeholder}
+          value=${args.value}
+          name=${args.name}
+          status=${args.status}
+          message=${args.message}
+          icon=${args.icon}
+          disabled=${args.disabled}
+          readonly=${args.readonly}
+          required=${args.required}
+          hasclearicon=${args.hasclearicon}
+          htmlid=${args.htmlid}
+          htmltitle=${args.htmltitle}
+          autocomplete=${args.autocomplete}
+        ></z-input>
+      `
+    }
   </Story>
-  <Story name="z-input type:radio">
-    {html`
-      <z-input
-        type="radio"
-        name="test-radio"
-        id="radio1"
-        label="${text('label', 'Radio Label')}"
-        labelafter="${boolean('labelafter', true)}"
-        disabled="${boolean('disabled', false)}"
-        value="${text('value', 'radio')}"
-        checked="${boolean('checked', false)}"
-      ></z-input>
-    `}
+  <Story
+    name="z-input textarea"
+    parameters={{
+      controls: {
+        include: [
+          "typeTextarea",
+          "label",
+          "ariaLabel",
+          "placeholder",
+          "value",
+          "name",
+          "status",
+          "message",
+          "disabled",
+          "readonly",
+          "required",
+          "htmlid",
+          "htmltitle",
+        ],
+      },
+    }}
+  >
+    {(args) =>
+      html`
+        <z-input
+          type=${args.typeTextarea}
+          label=${args.label}
+          aria-label=${args.ariaLabel}
+          placeholder=${args.placeholder}
+          value=${args.value}
+          name=${args.name}
+          status=${args.status}
+          message=${args.message}
+          disabled=${args.disabled}
+          readonly=${args.readonly}
+          required=${args.required}
+          htmlid=${args.htmlid}
+          htmltitle=${args.htmltitle}
+        ></z-input>
+      `
+    }
   </Story>
-  <Story name="z-input type:select">
-    {html`
-      <z-input
-        type="select"
-        label="${text('label', 'Select Label')}"
-        placeholder="Placeholder text"
-        status="${text('status', 'success')}"
-        message="${text('message', 'Helper text')}"
-        value="${text('value', 'select')}"
-        items='[{"id":"item_1","name":"first item","selected":false},{"id":"item_2","name":"second item","selected":true}]'
-      ></z-input>`}
+  <Story
+    name="z-input checkbox"
+    parameters={{
+      controls: {
+        include: [
+          "typeCheckbox",
+          "label",
+          "ariaLabel",
+          "labelPosition",
+          "name",
+          "disabled",
+          "readonly",
+          "required",
+          "checked",
+          "htmlid",
+          "htmltitle",
+        ],
+      },
+    }}
+  >
+    {(args) =>
+      html`
+        <z-input
+          type="${args.typeCheckbox}"
```

Comment:
non capisco questi control `typeQualcosa`. se servono solo a settare il type, puoi mettere `type` tra gli `exclude` dei `parameters` (in modo che non venga cambiato da chi sta usando storybook) e settare il type solo per la storia usando gli `args`: in questo caso la `Story` sarebbe
```jsx
<Story
  name="z-input checkbox"
  args={{
    type: 'checkbox',
  }}
  parameters={{
      ...
  }}
```

Reply:
Era un modo per mostrare sempre la tendina e far capire cosa potessi selezionare: quindi in caso di checkbox e radio non si cambia, invece per text/password/email/number è modificabile, e nella vista generale li selezioni tutti. 
Non ho trovato un modo per settare tutto sotto, questo mi sembrava il modo meno sporco.

Reply:
aah ho capito. se vuoi fare questa divisione delle option, puoi riusare di nuovo `argTypes` nella storia. per esempio in quella della input text puoi usare il `type` e mettere
```jsx
argTypes={{
  type: {
    options: ["text", "number", "password", "email"]
  }
}}
```